clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -576,16 +590,21 @@ private void addToCompletedReceives() {
                 KafkaChannel channel = entry.getKey();
                 if (!channel.isMute()) {
                     Deque<NetworkReceive> deque = entry.getValue();
-                    NetworkReceive networkReceive = deque.poll();
-                    this.completedReceives.add(networkReceive);
-                    this.sensors.recordBytesReceived(channel.id(), networkReceive.payload().limit());
+                    addToCompletedReceives(channel, deque, false);
```

Comment:
@hachikuji Thank you for the reviews. Agree that it is better to avoid the boolean flag. In fact, it  doesn't really need another method since it is just a single line loop during close(). Updated.